diff --git a/exe/json b/exe/json
index 9cca280..633de8e 160000
--- a/exe/json
+++ b/exe/json
@@ -1 +1 @@
-Subproject commit 9cca280a4d0ccf0c08f47a99aa71d1b0e52f8d03
+Subproject commit 633de8e44bbbc29383d72f8c920343ea5d1eaa21
diff --git a/exe/lns-removal-replan b/exe/lns-removal-replan
index b166024..fd2c59a 160000
--- a/exe/lns-removal-replan
+++ b/exe/lns-removal-replan
@@ -1 +1 @@
-Subproject commit b16602452ff20e9e161049812dce6d885319e1e4
+Subproject commit fd2c59aa167f94dfbe21971cbff2c7d52ace8ff2
diff --git a/exe/pbs-replan b/exe/pbs-replan
--- a/exe/pbs-replan
+++ b/exe/pbs-replan
@@ -1 +1 @@
-Subproject commit 35404c4b9af2e44e1478db03660dfd99b71e79e9
+Subproject commit 35404c4b9af2e44e1478db03660dfd99b71e79e9-dirty
diff --git a/exe/pp-replan b/exe/pp-replan
index 44dbcf0..531d643 160000
--- a/exe/pp-replan
+++ b/exe/pp-replan
@@ -1 +1 @@
-Subproject commit 44dbcf0edd63d9d44eed0ad87520b123ab7c63fe
+Subproject commit 531d643d416a2980f9b23fa1b385d4a0ea24e52d-dirty
